be/src/formats/csv/csv_reader.cpp

```diff
@@ -14,20 +14,532 @@
+        } else {
+            return Status::OK();
+        }
+    }
+}
+
+#define DOUBLE_CHECK_AVAILABLE()                \
+    if (_buff.available() < 1 && !notGetLine) { \
+        curState = NEWLINE;                     \
+        reachBuffEnd = true;                    \
+        goto newline_label;                     \
+    }
+
+#define READ_MORE()                        \
+    if (UNLIKELY(_buff.available() < 1)) { \
+        status = readMore(notGetLine);     \
+        if (!status.ok()) {                \
+            curState = NEWLINE;            \
+            goto newline_label;            \
+        }                                  \
+        DOUBLE_CHECK_AVAILABLE()           \
+    }
+
+#define READ_MORE_ENCLOSE()                \
+    if (UNLIKELY(_buff.available() < 1)) { \
+        status = readMore(notGetLine);     \
+        if (!status.ok()) {                \
+            is_enclose_field = true;       \
+            curState = NEWLINE;            \
+            goto newline_label;            \
+        }                                  \
+        DOUBLE_CHECK_AVAILABLE()           \
+    }
+
+Status CSVReader::more_lines() {
+    if (UNLIKELY(_limit > 0 && _parsed_bytes > _limit)) {
+        return Status::EndOfFile("Reached limit");
+    }
+    Status status = Status::OK();
+    status = buffInit();
+    if (!status.ok()) {
+        return status;
+    }
+    _escape_data.clear();
+    _escape_pos.clear();
+    ParseState curState = START;
+    ParseState preState = curState;
+    // parsed_start and parsed_end record a line at the start and end of the buff.
+    size_t parsed_start = _buff.position_offset();
+    size_t parsed_end = 0;
+    // If field_start is initialized to the maximum value, the START state is not entered.
+    size_t field_start = std::string::npos;
+    size_t field_end = 0;
+    bool is_enclose_field = false;
+    bool is_escape_field = false;
+    bool notGetLine = true;
+    bool reachBuffEnd = false;
+    _fields.clear();
+    while (true) {
+        // At the end of a row, or the end of a field, no new data is read.
+        if (LIKELY(curState != NEWLINE && curState != DELIMITER)) {
+            READ_MORE()
+        }
+        // Advance to the next state each time based on the current state + the current character stream.
+        switch (curState) {
+        case START:
+            // Trim space is enabled and the leading space is skipped in the START state
+            if (UNLIKELY(_parse_options.trim_space)) {
+                while (*(_buff.position()) == ' ') {
+                    _buff.skip(1);
+                    READ_MORE()
+                }
+            }
+            field_start = _buff.position_offset();
+
+            // newline
+            if (UNLIKELY(is_row_delimiter(notGetLine))) {
+                curState = NEWLINE;
+                break;
+            }
+
+            // delimiter
+            if (UNLIKELY(is_column_separator(notGetLine))) {
```

Comment:
Because the column separator is now been used in stream load.
To do better, we should be unified to column_delimiter/row_delimiter.